Explorer/Assets/DCL/Navmap/NavmapSearchBarController.cs

```diff
@@ -1,186 +1,225 @@
 using Cysharp.Threading.Tasks;
-using DCL.AssetsProvision;
 using DCL.Input;
 using DCL.Input.Component;
-using DCL.PlacesAPIService;
+using DCL.MapRenderer.MapLayers.Categories;
+using DCL.Navmap.ScriptableObjects;
 using DCL.UI;
-using DCL.WebRequests;
 using System;
-using System.Linq;
 using System.Threading;
-using UnityEngine;
 using Utility;
 
 namespace DCL.Navmap
 {
     public class NavmapSearchBarController : IDisposable
     {
-        private const string PREVIOUS_SEARCHES_KEY = "previous_searches";
-        private const int MAX_PREVIOUS_SEARCHES = 5;
-
-        public event Action<string> OnResultClicked;
-        public event Action OnSearchTextChanged;
+        public event Action<bool>? TogglePanel;
 
         private readonly SearchBarView view;
-        private readonly IPlacesAPIService placesAPIService;
         private readonly HistoryRecordPanelView historyRecordPanelView;
-        private readonly SearchResultPanelController searchResultPanelController;
+        private readonly SearchFiltersView searchFiltersView;
         private readonly IInputBlock inputBlock;
+        private readonly INavmapBus navmapBus;
+        private readonly CategoryMappingSO categoryMappingSO;
 
-        private CancellationTokenSource cts;
+        private CancellationTokenSource? searchCancellationToken;
+        private CancellationTokenSource? backCancellationToken;
         private bool isAlreadySelected;
-        private string[] previousSearches;
-        private string previousSearchesString;
-        private string playerPrefsPreviousSearches;
+        private NavmapSearchPlaceFilter currentPlaceFilter = NavmapSearchPlaceFilter.All;
+        private NavmapSearchPlaceSorting currentPlaceSorting = NavmapSearchPlaceSorting.MostActive;
+        private string? currentCategory;
+        private string currentSearchText = string.Empty;
+
+        public bool Interactable
+        {
+            get => view.inputField.interactable;
+            set => view.inputField.interactable = value;
+        }
 
         public NavmapSearchBarController(
             SearchBarView view,
-            SearchResultPanelView searchResultPanelView,
             HistoryRecordPanelView historyRecordPanelView,
-            IPlacesAPIService placesAPIService,
-            FloatingPanelView floatingPanelView,
-            IWebRequestController webRequestController,
-            IInputBlock inputBlock
-        )
+            SearchFiltersView searchFiltersView,
+            IInputBlock inputBlock,
+            ISearchHistory searchHistory,

```

Comment:
You are not using search history at all. Therefore, all the implemented classes can be safely deleted; or you need to add the usage